src/engines/kotor/gui/chargen/chargenname.cpp

```diff
@@ -36,23 +39,41 @@ CharacterGenerationNameMenu::CharacterGenerationNameMenu(CharacterGenerationInfo
 
 	load("name");
 
+	_name = "";
+
 	addBackground(kBackgroundTypeMenu);
 
-	getLabel("NAME_BOX_EDIT")->setFont("fnt_d16x16b");
-	getLabel("NAME_BOX_EDIT")->setText("_");
+	_nameLabel = getLabel("NAME_BOX_EDIT");
+	_nameLabel->setFont("fnt_d16x16b");
+	_nameLabel->setText("_");
+
+	EventMan.enableTextInput(true);
 }
 
 void CharacterGenerationNameMenu::callbackActive(Widget &widget) {
 	if (widget.getTag() == "BTN_BACK") {
 		_returnCode = 1;
+		EventMan.enableTextInput(false);
 		return;
 	}
 
 	if (widget.getTag() == "END_BTN") {
 		_returnCode = 1;
 		accept();
+		EventMan.enableTextInput(false);
+		return;
+	}
+}
+
+void CharacterGenerationNameMenu::callbackTextInput(const Common::UString &text) {
+	// The name should not be longer than 18 letters (according to the original game).
+	if (_name.size() == 18) {
 		return;
 	}
+
+	_name += text;
+	_info.setName(_name);
+	_nameLabel->setText(_name + "_");
```

Comment:
That is missing the ability to delete characters again with backspace, though.

I guess you also need a callbackKeyDown() for that.

Reply:
Yes, this is already planned, but for now I assumed, it would be enough to enable typing. At I have to design a new callback for keys, but this was planned for another pull request.

Reply:
Eh, do that other pull request first, please, and then revisit this pull request. I think it would be best for this PR to be more feature-complete before merging it. :)